pkg/frontend/querymiddleware/split_and_cache_test.go

```diff
@@ -407,6 +425,16 @@ func TestSplitAndCacheMiddleware_ResultsCache_ShouldNotCacheRequestEarlierThanMa
 			expectedDownstreamStartTime: fiveMinutesAgo,
 			expectedDownstreamEndTime:   now,
 			expectedCachedResponses:     nil,
+			expectedMetrics: `
+				# HELP cortex_frontend_query_result_cache_skipped_total Total number of times a query was not cacheable because of a reason.
+				# TYPE cortex_frontend_query_result_cache_skipped_total counter
+				cortex_frontend_query_result_cache_skipped_total{reason="modifiers-not-cacheable"} 0
+				cortex_frontend_query_result_cache_skipped_total{reason="too-new"} 2
+				cortex_frontend_query_result_cache_skipped_total{reason="unaligned-request"} 0
+				# HELP cortex_frontend_split_queries_total Total number of underlying query requests after the split by interval is applied.
+				# TYPE cortex_frontend_split_queries_total counter
+				cortex_frontend_split_queries_total 0
```

Comment:
The value here is zero. As you can see there's a discrepancy between `cortex_frontend_query_result_cache_skipped_total` (sum is 2) with `cortex_frontend_split_queries_total`. `cortex_frontend_split_queries_total` is used in your new dashboard panel to compute the %, but the 2 metrics are not safe to compare. To keep it safe, you should add yet another metric `cortex_frontend_query_result_cache_attempted_total` which is tracked for each partial query checked whether we could cache it or not, and then use it in the dashboard panel.